[docs] Migrate button demos to base (#31395)
mui/material-ui: docs/data/base/components/button/button.md

---
product: base
title: Unstyled React button
components: ButtonUnstyled
githubLabel: 'component: button'
waiAria: https://www.w3.org/TR/wai-aria-practices/#button
---

# Unstyled button

> <p class="description">Buttons allow users to take actions, and make choices, with a single tap.</p>

Reviewer: ```suggestion
<p class="description">Buttons allow users to take actions and make choices with a single tap.</p>
```

@samuelsycamore I'm not sure about the second comma, though. Could you please verify?

## Basic usage

```js
import ButtonUnstyled from '@mui/base/ButtonUnstyled';
```

{{"demo": "UnstyledButtonsSimple.js"}}

## Customizing the root element

By default, the `ButtonUnstyled` component renders a native `button` HTML element.
You can override this by setting the `component` or `components.Root` prop.
If you provide a non-interactive element, such as a `span`, the `ButtonUnstyled` component will automatically add the necessary accessibility attributes.

{{"demo": "UnstyledButtonsSpan.js"}}

Compare the attributes on the `span` with the `button` from the previous demo.

## Complex customization

> In addition to HTML elements, you can also use SVGs with the 'ButtonUnstyled` component.

Reviewer: ```suggestion
In addition to HTML elements, you can also use SVGs with the `ButtonUnstyled` component.
```

...or remove the backticks. We should decide and be consistent about surrounding the component name with backticks. I'd say it's not necessary to show component names as code. I would use them only for prop and parameter names and values.
cc @samuelsycamore

## Basic usage

```js
import ButtonUnstyled from '@mui/base/ButtonUnstyled';
```

{{"demo": "UnstyledButtonsSimple.js"}}

## Customizing the root element

By default, the `ButtonUnstyled` renders a native `button` element.
You are free to override this by setting the `component` or `components.Root` prop.
If a non-interactive element (such as a span) is provided this way, the `ButtonUnstyled` will take care of adding accessibility attributes.

{{"demo": "UnstyledButtonsSpan.js"}}

> Compare the attributes on the span with the button from the previous demo.

Reviewer 2: ```suggestion
Compare the attributes on the `span` with the `button` from the previous demo.
```

Reviewer 2: Also, should this sentence go above the demo? I think it's preferable to tell the reader what to look for in the demo before they see it.

## Complex customization

> You are not limited to using HTML elements for the button structure.
> SVG elements, even with complex structure, are equally acceptable.

Reviewer 2: These two sentences can be condensed into one:

```suggestion
In addition to HTML elements, you can also use SVGs with the 'ButtonUnstyled` component.
```

{{"demo": "UnstyledButtonCustom.js"}}

> ## useButton hook

Reviewer 2: ```suggestion
## The useButton hook
```

Author: Does "The" really need is this case?

```js
import { useButton } from '@mui/base/ButtonUnstyled';
```

> If you need to use Button's functionality in another component, you can use the `useButton` hook.

Reviewer 2: Rearrange the sentence to place the main subject up front, and be more clear about "Button."

```suggestion
The `useButton` hook lets you use the functionality of `ButtonUnstyled` in other components.
```

> It returns props to be placed on a custom button element and fields representing the internal state of the button.

Reviewer 2: Minor style tweak to make it more clear that there are two kinds of things being returned.

```suggestion
It returns props to be placed on a custom button element, along with fields representing the internal state of the button.
```

> The `useButton` hook requires the ref of the element it'll be used on.

Reviewer 2: I believe "ref" refers to `ref` as in an element of the code, so I think it should get back ticks. Also, let's keep it in the present tense.

```suggestion
The `useButton` hook requires the `ref` of the element it's used on.
```
